pytorch_lightning/callbacks/progress/base.py

```diff
@@ -47,52 +47,57 @@ def on_train_batch_end(self, trainer, pl_module, outputs, batch_idx):
     """
 
     def __init__(self):
-
-        self._trainer = None
-        self._train_batch_idx = 0
-        self._val_batch_idx = 0
-        self._test_batch_idx = 0
-        self._predict_batch_idx = 0
+        self._trainer: Optional["pl.Trainer"] = None
 
     @property
-    def trainer(self):
+    def trainer(self) -> "pl.Trainer":
         return self._trainer
 
     @property
     def train_batch_idx(self) -> int:
-        """The current batch index being processed during training.
+        """The number of batches processed during training.
 
         Use this to update your progress bar.
         """
-        return self._train_batch_idx
+        if self.trainer is None:
+            return 0
+        return self.trainer.fit_loop.epoch_loop.batch_progress.current.processed
```

Comment:
I wonder if this will break the kfold example.

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/9b873dcfcc9f060a351d559c30a0250efbd8a6b8/pl_examples/loop_examples/kfold.py#L279-L280
https://github.com/PyTorchLightning/pytorch-lightning/blob/9b873dcfcc9f060a351d559c30a0250efbd8a6b8/pl_examples/loop_examples/kfold.py#L173-L176

Reply:
It doesn't because of

https://github.com/PyTorchLightning/pytorch-lightning/blob/3a2df4f75d104b0ecf0af9b35e8ebd535a99a591/pl_examples/loop_examples/kfold.py#L236-L240

Reply:
ah! that's quite neat!